dataset/bins.cpp

```diff
@@ -405,4 +405,53 @@ Dataset sum(const Dataset &d) {
   return apply_to_items(d, [](auto &&... _) { return buckets::sum(_...); });
 }
 
+Variable mean(const Variable &data) {
+  auto type = variable::variableFactory().elem_dtype(data);
+  type = type == dtype<bool> ? dtype<int64_t> : type;
+  const auto unit = variable::variableFactory().elem_unit(data);
+
+  Variable means;
+  if (variable::variableFactory().hasVariances(data))
+    means = Variable(type, data.dims(), unit, Values{}, Variances{});
+  else
+    means = Variable(type, data.dims(), unit, Values{});
```

Comment:
Same code as in `sum` above, extract into helper function in an anonymous namespace?